src/rocksdb.rs

```diff
@@ -250,17 +293,27 @@ impl<D> DBIterator<D> {
     }
 
     pub fn kv(&self) -> Option<(Vec<u8>, Vec<u8>)> {
-        if self.valid() {
+        let valid = self.checked_valid().expect("checked_valid fail");
+        if valid {
             Some((self.key().to_vec(), self.value().to_vec()))
         } else {
             None
         }
     }
 
+    #[deprecated]
     pub fn valid(&self) -> bool {
```

Comment:
Maybe better to keep the `valid()` interface to be aligned with the RocksDB C/C++ API.